.github/workflows/build_dev_man.yml

```diff
@@ -1,59 +1,45 @@
-name: Build and push for dev (Manually)
+name: Build and push for dev (Self-hosted)
 
 env:
   MODULES_REGISTRY: ${{ vars.DEV_REGISTRY }}
-  CI_COMMIT_REF_NAME: ${{ github.event.inputs.tag }}
-  #  CI_COMMIT_REF_NAME: ${{ github.ref_name }}
+  CI_COMMIT_REF_NAME: ${{ github.ref_name }}
   MODULES_MODULE_NAME: ${{ vars.MODULE_NAME }}
   MODULES_MODULE_SOURCE: ${{ vars.DEV_MODULE_SOURCE }}
   MODULES_REGISTRY_LOGIN: ${{ vars.DEV_MODULES_REGISTRY_LOGIN }}
   MODULES_REGISTRY_PASSWORD: ${{ secrets.DEV_MODULES_REGISTRY_PASSWORD }}
 
 on:
-  workflow_dispatch:
-    inputs:
-
-      tag:
-        description: "The module's tag, which must pr654 For example: pr56"
-        type: string
-        required: true
-#  pull_request:
-#    types: [ opened, edited, synchronize ]
+  pull_request:
+    types: [ labeled ]
 #  push:
 #    branches:
 #      - main
+
+defaults:
+  run:
+    shell: bash
+
 concurrency:
-  group: ${{ github.workflow }}-${{ github.event.number || github.event.inputs.tag }}
+  group: ${{ github.workflow }}-${{ github.event.number || github.ref }}
   cancel-in-progress: true
 
 jobs:
-  tag_validation:
-    runs-on: [self-hosted, test-vm-runner]
-    steps:
-      - name: Validation for tag
-        run: |
-          echo ${{ github.event.inputs.tag }} | grep -P '^pr\d+$'
-        shell: bash
-
   show_dev_manifest:
+    if: ${{ github.event.label.name == 'build/self-hosted/virt-runners' }}
```

Comment:
It can be optimized a little. Create a new, first job, check for label there, and create output 'should_run'. This job should be added as a dependency in all other jobs. `if: long js snippet with hardcoded label` become `if: ${{ jobs.check_label.outputs.should_run }}`